src/methods.jl

```diff
@@ -127,13 +155,23 @@ In-place version of [`flat_correct`](@ref)
 # See Also
 [`flat_correct`](@ref)
 """
-function flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; norm_value = mean(flat_frame))
+function flat_correct!(frame::AbstractArray{T}, flat_frame::AbstractArray; norm_value = mean(flat_frame)) where T
     norm_value <= 0 && error("norm_value must be positive")
-    frame ./= (flat_frame ./ norm_value)
+    @. frame = convert_value(T, frame / (flat_frame / norm_value))
```

Comment:
no, we shouldn't do this. 

The reason this was introduced for `subtract_overscan`, as we discussed on Zulip, was because we don't yet have an interface for specifying *how* to model the overscan region.

In this case we need to be much more exact, so we can't just blindly round and convert floats to ints, for instance. 

In this case, we _should_ just let the code error if the left hand side (`frame`) does not have a compatible type. Using the `!` version shifts this responsibility to the user!

However, for the non `!` versions, we'll want to convert the frame (and are allowed to!) before passing it on-
```julia
function flat_correct(frame::AbstractArray{T}, flat::AbstractArray{S}, kwargs...) where T,S
    V = float(T, S)
    flat_correct!(V.(frame), flat, kwargs...)
end
```

Reply:
True, rounding off can lead to loss of important data! I will make the changes and push the code!

Reply:
I had a small typo, should be
```julia
V = float(promote_type(T, S))
```